aiida/cmdline/commands/cmd_group.py

```diff
@@ -59,33 +61,27 @@ def group_remove_nodes(group, nodes, clear, force):
 
 @verdi_group.command('delete')
 @arguments.GROUP()
-@options.GROUP_CLEAR(help='Remove all nodes before deleting the group itself.')
+@options.GROUP_CLEAR(
+    help='Remove all nodes before deleting the group itself.' +
+    ' [deprecated: No longer has any effect. Will be removed in 2.0.0]'
+)
 @options.FORCE()
 @with_dbenv()
 def group_delete(group, clear, force):
     """Delete a group.
 
-    Note that a group that contains nodes cannot be deleted if it contains any nodes. If you still want to delete the
-    group, use the `-c/--clear` flag to remove the contents before deletion. Note that in any case, the nodes themselves
-    will not actually be deleted from the database.
+    Note that this command only deletes groups - nodes contained in the group will remain untouched.
     """
     from aiida import orm
 
     label = group.label
 
-    if group.count() > 0 and not clear:
-        echo.echo_critical((
-            'Group<{}> contains {} nodes. Pass `--clear` if you want to empty it before deleting the group'.format(
-                label, group.count()
-            )
-        ))
+    if clear:
+        warnings.warn('`--clear` is deprecated and no longer has any effect.', AiidaDeprecationWarning)  # pylint: disable=no-member
 
```

Comment:
Perhaps we can still add a warning in case the group is not empty and the `force` flag isn't set?

```suggestion
    if group.count() > 0 and not force:
        echo.echo_warning(f'Group <{label}> contains {group.count()} nodes.')

```

Reply:
N.B.: `.count()` can potentially be expensive, so it's better to call it once and assign to variable

Reply:
thanks @mbercx - I added it but then I had second thoughts 

 * Warning people how many nodes there are inside the group can give the impression that we're deleting nodes here (while it is just the group that is being deleted)
 * As @sphuber mentions, it can slow down the command

I.e. unless you feel strongly about this, I prefer to leave the code as it is

Reply:
> Warning people about how many nodes there are inside the group can give the impression that we're deleting nodes here (while it is just the group that is being deleted)

That's a good point, I hadn't thought about that. I suppose we can still rephrase the warning, but it's not critical.

> As @sphuber mentions, it can slow down the command

I actually tested this for one of our larger groups on theossrv5, and it's not _that_ time consuming:

```Python
In [10]: g.count()                                                                                                                     
Out[10]: 61134

In [11]: timeit g.count()                                                                                                              
174 ms ± 1.39 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

But, I do not feel super strongly about it, so feel free to merge!

Reply:
this may be on the sqlalchemy backend, where I remember trying to optimize the shit out of it because it was really slow at some point and it shouldn't be. could be different on Django, could be also ok though, not sure. But I am also fine with leaving it as is as @ltalirz suggests. Let's get this thing merged.